src/gather/problem_description.cpp

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/errors.hpp>
+#include <miopen/gather/problem_description.hpp>
+#include <miopen/tensor.hpp>
+
+#include <sstream>
+
+namespace miopen {
+
+namespace gather {
+
+bool FwdProblemDescription::IsSameType() const
+{
+    if(inputDesc.GetType() != outputDesc.GetType())
+    {
+        return false;
+    }
+
+    return true;
+}
+
+NetworkConfig FwdProblemDescription::MakeNetworkConfig() const
+{
+    std::ostringstream ss;
+
+    ss << "gather";
+    ss << "dtype" << inputDesc.GetType();
+    ss << "output_size" << outputDesc.GetElementSize();
+    ss << "mode " << gatherDesc.getMode();
+    ss << "dim " << gatherDesc.getDim();
```

Comment:
redundant because you dont use dim to specify solver

Reply:
updated.